NetCDF2zarrPhysModel.py

```diff
@@ -0,0 +1,207 @@
+#------------------------------------------------------------------------------------------------------------------
+# Convertion from netcdf to zarr_store using xarray and pandas libraries
+# --------------------------------------------------------------------------------------------------------------------
+#
+#   USAGE
+#
+#  python3 NetCDF2zarrPhysModel.py start_time number_batches_append batch_size
+#  or
+#  python3 NetCDF2zarrPhysModel.py --basedir base_dir_name  start_time number_batches_append batch_size
+#
+#  start_time            - begining time stamp given by user
+#  number_batches_append - how many time steps to keep
+#  batch_size            - size of time in seconds
+#
+#  start_time is provided as a string in format "dd/mm/yy HH:MM:SS"
+#
+#-----------------------------------------------------------------------------------------------------------------------'''
+
+import argparse
+from dask.distributed import Client, progress, LocalCluster
+import pandas as pd
+from pathlib import Path
+import xarray as xr
+import numpy as np
+import numcodecs
+import zarr
+from datetime import datetime
+import time
+import sys
+
+# Define time string formats:
+format_short = "ddMMyyyyHHmmSS"
+# format for time string extended
+format_extended = "dd/MM/yy HH:mm:SS"
+
+def mkdate(datestr):
+   # function make time string for argument parser
+   try:
+    return datestr
+   except ValueError:
+    raise argparse.ArgumentTypeError(datestr + 'time start is not in a proper format "dd/MM/yy HH:mm:SS" ')
+
+def convert_time_formats(t_short):
+   # convert from "ddMMyyyyHHmmSS" to "dd/mm/yy HH:MM:SS"
+   t_extended = t_short[0:2] + "/" + t_short[2:4] + "/" + \
+                t_short[6:8] + " " + t_short[8:10] + ":" + \
+                t_short[10:12] + ":" + t_short[12:14]
+   return t_extended
+
+def convert_secs_date(time_secs,time_format):
+   # convert from seconds to time given in format "dd/mm/yy HH:MM:SS"
+   dt_object = datetime.fromtimestamp(time_secs)
+   str_time = dt_object.strftime("%d/%m/%y %H:%M:%S")
+   return str_time
+
+def convert_to_secs(tt,time_time_format):
+    # Convertion from time in format dd/MM/yy HH:mm:SS to time in seconds
+    # tt - should be given in format "dd/mm/yy HH:MM:SS"
+    obj1 = time.strptime(tt,"%d/%m/%y %H:%M:%S")
+    time_secs = time.mktime(obj1)
+    return time_secs
+
+def extract_time_stamp_from_name(name_netcdf_file, time_format):
+    # get the prefix name and time intervals from netcdf file name
+    # the time format string is "ddMMyyyyHHmmSS"
+    # the netcdf file naming : {netcdf_file_prefix}_ddMMyyyyHHmmSS_ddMMyyyyHHmmSS.nc
+    len_stamp=len(time_format)
+    netcdf_file_prefix = name_netcdf_file[0:-2*len_stamp-2]
+    time_start = name_netcdf_file[-2*len_stamp-1:-len_stamp-1]
+    time_end = name_netcdf_file[-len_stamp:]
+    return time_start, time_end, netcdf_file_prefix
+
+
+def create_or_append_zarr(names_netcdf, zarr_store, start_time, \
```

Comment:
I thinks we can simplify this for our use case. What we want is something along the lines of
```python
# open all netcdf files into a large virtual dataset
data_set = xarray.open_mfdataset(names_netcdf)

# clear encoding
# (This is necessary, because xarray tries to keep track of the netCDF
# specifics which are not needed for Zarr conversion but which can
# lead to problems, with, e.g., chunksizes being retained in the encoding 
# even if they don't match the underlying dask arrays any more.)
for varname in data_set.data_vars.keys():
    data_set[varname].encoding = {}

# Convert to zarr
data_set.to_zarr(
    name_of_zarr_store,  # path to the zarr store (e.g. "./GLOBAL_ANALYSIS_FORECAST_WAVE_001_027-TDS.zarr"
    mode="w"  # careful here: This will delete whatever is in name_of_zarr_store
)
```